File: C++/21.cpp
#include <iostream> 
using namespace std;  
int GCD(int a, int b) 
{ 
	// Everything divides 0 
	if (a == 0) 
	return b; 
	if (b == 0) 
	return a; 

	// both are equal
	if (a == b) 
		return a; 

	// a is greater 
	if (a > b) 
		return GCD(a-b, b); 
	//b is greater
	return GCD(a, b-a); 
} 


int main() 
{ 
	int x, y;
	cout<<"Enter the two numbers "<<endl;
	cin>>x>>y;
	cout<<"GCD of "<<x<<" and "<<y<<" is "<<GCD(x, y); 
	return 0; 
} 

File: Python/21.py
def GCD(a,b):
    if b==0:
        return a
    r=a%b
    return GCD(b,r)
numbers=input("Enter two numbers:")
nums=numbers.split()
a=int(nums[0])
b=int(nums[1])
print("GCD of 2 numbers:"GCD(a,b))
